processor/src/main/kotlin/com/syouth/kmapper/processor/convertors/DataClassTypeConverter.kt

```diff
@@ -22,7 +22,7 @@ internal class DataClassTypeConverter(
 ) : TypeConvertor {
     override fun isSupported(from: KSType?, to: KSType, targetPath: PathHolder?): Boolean {
         if (from == null) return false
-        return from.isDataClass() && to.isDataClass() &&
+        return (from.isDataClass() || to.isDataClass()) &&
```

Comment:
Here, it should work for all 4 cases as well.

Reply:
At the moment, conversion works if at least to or from is of type `data class`. Should this converter be restricted to convert `data class`es only or should it be extended to be capable of convert `class` to `class` as well?
Replacing the check with `return checkDifferentTypesNullabilitySufficient(from, to)` breaks the `testload` build with
```
> Task :testload:kspKotlin FAILED
e: [ksp] java.lang.IllegalStateException: Data class should have primary constructor
        at com.syouth.kmapper.processor.base.UtilsKt.buildMappingTable(Utils.kt:55)
        at com.syouth.kmapper.processor.convertors.DataClassTypeConverter.buildCodeBlockForDifferentTypes(DataClassTypeConverter.kt:48)
```
I haven't investigated in the root cause yet, though.

Reply:
I think, I've found a solution for this.

Reply:
I see. I think it will work as a temporary solution but proper handling probably will take more changes(I want to keep it as independent as possible). But I haven't even invented it yet so it will do the job :).